naga/src/front/wgsl/lower/mod.rs

```diff
@@ -2208,6 +2209,15 @@ impl<'source, 'temp> Lowerer<'source, 'temp> {
                     .collect::<Result<Vec<_>, _>>()?;
 
                 let has_result = ctx.module.functions[function].result.is_some();
+                let must_use = ctx.module.functions[function]
+                    .result
+                    .as_ref()
+                    .is_some_and(|result| result.must_use);
+
+                if must_use && is_statement {
+                    return Err(Error::FunctionMustUseUnused(span));
```

Comment:
`function.span` (where `function` is the argument to `call()`) is a better span to use than `span`. However, the argument `function` is being shadowed by this pattern match arm on line 2187. We can either rename `function` in the match arm, or create a new variable `function_span` prior to the match block.

This additionally means you don't have to `.split_once('(')` the span in error.rs